WordPress/Classes/ViewRelated/Comments/CommentDetailInfoViewController.swift

```diff
@@ -100,24 +104,23 @@ extension CommentDetailInfoViewController: UITableViewDelegate {
     }
 }
 
-// MARK: - DrawerPresentable
-extension CommentDetailInfoViewController: DrawerPresentable {
-    var collapsedHeight: DrawerHeight {
-        .intrinsicHeight
-    }
-
-    var allowsUserTransition: Bool {
-        false
-    }
-
-    var compactWidth: DrawerWidth {
-        .maxWidth
-    }
-}
-
-// MARK: - ChildDrawerPositionable
-extension CommentDetailInfoViewController: ChildDrawerPositionable {
-    var preferredDrawerPosition: DrawerPosition {
-        .collapsed
+extension CommentDetailInfoViewController {
+    func show(from presentingViewController: UIViewController, sourceView: UIView) {
+        let navigationController = UINavigationController(rootViewController: self)
+        if UIDevice.isPad() {
+            navigationController.modalPresentationStyle = .popover
+            navigationController.popoverPresentationController?.sourceView = sourceView
+            navigationController.popoverPresentationController?.permittedArrowDirections = [.up]
+        } else {
+            self.navigationItem.leftBarButtonItem = UIBarButtonItem(title: SharedStrings.Button.close, primaryAction: UIAction { [weak presentingViewController] _ in
+                presentingViewController?.dismiss(animated: true)
+            })
```

Comment:
I feel like it makes better sense to put this code into the `if let sheet` block below.

Reply:
If it's presented modally with the default settings, it still needs a "Cancel" button.

Reply:
Ah of course. I misread the code...